appium/webdriver/webdriver.py

```diff
@@ -232,12 +232,23 @@ def __init__(
         if isinstance(command_executor, str):
             command_executor = AppiumConnection(command_executor, keep_alive=keep_alive)
 
+        if desired_capabilities is not None:
+            warnings.warn(
+                'desired_capabilities argument is deprecated and will be removed in future versions. '
+                'Use options instead.'
+            )
+        # TODO: Remove the fallback after desired_capabilities removal
+        dst_options = (
+            AppiumOptions().load_capabilities(desired_capabilities)
+            if desired_capabilities is not None and options is None
+            else options
+        )
+
         super().__init__(
             command_executor=command_executor,
-            desired_capabilities=desired_capabilities,
             browser_profile=browser_profile,
             proxy=proxy,
```

Comment:
Maybe we should remove browser_profile and proxy as well?
https://github.com/SeleniumHQ/selenium/pull/12030/files#diff-c87c150ed2a08f5293db58a06da8bf7f14c3a5582586c6270c1be661dfdd0985L205

Reply:
done